src/resizing.jl

```diff
@@ -0,0 +1,158 @@
+"""
+`imgr = restrict(img[, region])` performs two-fold reduction in size
+along the dimensions listed in `region`, or all spatial coordinates if
+`region` is not specified.  It anti-aliases the image as it goes, so
+is better than a naive summation over 2x2 blocks.
+"""
+restrict(img::AbstractArray, ::Tuple{}) = img
+
+typealias RegionType Union{Dims,Vector{Int}}
+
+function restrict(A::AbstractArray, region::RegionType=coords_spatial(A))
+    for dim in region
+        A = _restrict(A, dim)
+    end
+    A
+end
+
+function _restrict{T,N}(A::AbstractArray{T,N}, dim::Integer)
+    if size(A, dim) <= 2
+        return A
+    end
+    newsz = ntuple(i->i==dim?restrict_size(size(A,dim)):size(A,i), Val{N})
+    out = Array{typeof(A[1]/4+A[2]/2),N}(newsz)
+    restrict!(out, A, dim)
+    out
+end
+
+# out should have efficient linear indexing
```

Comment:
Ugh, I'd forgotten that `restrict` still used the old `Base.Cartesian` rather than `CartesianIndex`. Keep it like it is now, but someday I should change this.